mev-programs/programs/priority-fee-distribution/src/lib.rs

```diff
@@ -0,0 +1,790 @@
+        let merkle_root_upload_config = &mut ctx.accounts.merkle_root_upload_config;
+        merkle_root_upload_config.override_authority = authority;
+        merkle_root_upload_config.original_upload_authority = original_authority;
+        merkle_root_upload_config.bump = ctx.bumps.merkle_root_upload_config;
+        Ok(())
+    }
+
+    pub fn update_merkle_root_upload_config(
+        ctx: Context<UpdateMerkleRootUploadConfig>,
+        authority: Pubkey,
+        original_authority: Pubkey,
+    ) -> Result<()> {
+        // Call the authorize function
+        UpdateMerkleRootUploadConfig::auth(&ctx)?;
+
+        // Update override authority
+        let merkle_root_upload_config = &mut ctx.accounts.merkle_root_upload_config;
+        merkle_root_upload_config.override_authority = authority;
+        merkle_root_upload_config.original_upload_authority = original_authority;
+
+        Ok(())
+    }
+
+    pub fn migrate_tda_merkle_root_upload_authority(
+        ctx: Context<MigrateTdaMerkleRootUploadAuthority>,
+    ) -> Result<()> {
+        let distribution_account = &mut ctx.accounts.priority_fee_distribution_account;
+        // Validate TDA has no MerkleRoot uploaded to it
+        if distribution_account.merkle_root.is_some() {
+            return Err(InvalidTdaForMigration.into());
+        }
+        // Validate the TDA key is the acceptable original authority (i.e. the original Jito Lab's authority)
+        if distribution_account.merkle_root_upload_authority
+            != ctx
+                .accounts
+                .merkle_root_upload_config
+                .original_upload_authority
+        {
+            return Err(InvalidTdaForMigration.into());
+        }
+
+        // Change the TDA's root upload authority
+        distribution_account.merkle_root_upload_authority =
+            ctx.accounts.merkle_root_upload_config.override_authority;
+
+        Ok(())
+    }
+
+    pub fn transfer_priorty_fee_tips(
+        ctx: Context<TransferPriortyFeeTips>,
+        lamports: u64,
+    ) -> Result<()> {
+        let epoch = Clock::get()?.epoch;
+        // Valdiate the PFDA is in the current epoch
+        require!(
+            ctx.accounts
+                .priority_fee_distribution_account
+                .epoch_created_at
+                == epoch,
+            ErrorCode::AccountValidationFailure
+        );
+        // Transfer requested lamports from From to PFDA
+        let ix = solana_program::system_instruction::transfer(
+            ctx.accounts.from.key,
+            &ctx.accounts.priority_fee_distribution_account.key(),
+            lamports,
+        );
+        solana_program::program::invoke(
+            &ix,
+            &[
+                ctx.accounts.from.to_account_info(),
+                ctx.accounts
+                    .priority_fee_distribution_account
+                    .to_account_info(),
+            ],
+        )
+        .map_err(Into::into)
+    }
```

Comment:
I tried mutating the lamports but ran into [this error](https://github.com/anza-xyz/agave/blob/master/transaction-context/src/lib.rs#L826-L829) _An account not owned by the program cannot have its balance decrease_. Didn't know this off the top of my head, but I guess in this case we have to use the system program to transfer.